flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/CatalogManager.java

```diff
@@ -415,6 +415,37 @@ public CatalogBaseTable getTable() {
 			});
 	}
 
+	/**
+	 * Returns an array of names of all views(both temporary and permanent) registered in
+	 * the namespace of the current catalog and database.
+	 *
+	 * @return names of all registered views
+	 */
+	public Set<String> listViews() {
+		return listViews(getCurrentCatalog(), getCurrentDatabase());
+	}
+
+	/**
+	 * Returns an array of names of all views(both temporary and permanent) registered in
+	 * the namespace of the current catalog and database.
+	 *
+	 * @return names of registered views
+	 */
+	public Set<String> listViews(String catalogName, String databaseName) {
+		Catalog currentCatalog = catalogs.get(getCurrentCatalog());
+
+		try {
+			return Stream.concat(
+				currentCatalog.listViews(getCurrentDatabase()).stream(),
+				listTemporaryTablesInternal(catalogName, databaseName)
+					.filter(e -> e.getValue() instanceof CatalogView)
+					.map(e -> e.getKey().getObjectName())
+			).collect(Collectors.toSet());
```

Comment:
add a method `listTemporaryViews(String catalogName, String databaseName)`, `listTemporaryViews()` method and here can use this method

Reply:
IMO the private `listTemporaryTablesInternal` is enough. or we should add `listTemporaryTables(String catalogName, String databaseName)` for `listTemporaryTables()`. That's  probably a bit superfluous.

Reply:
`listTemporaryTablesInternal` already exists